clients/go/internal/containersuite/containerSuite.go

```diff
@@ -57,6 +59,35 @@ func (s zeebeWaitStrategy) WaitUntilReady(ctx context.Context, target wait.Strat
 		return err
 	}
 
+	defer func() {
+		_ = zbClient.Close()
+	}()
+
+	finishedChan := make(chan error, 1)
+	go s.waitForTopology(finishedChan, zbClient)
+
+	select {
+	case err = <-finishedChan:
+		return err
+	case <-time.After(utils.DefaultContainerWaitTimeout):
+		reader, err := target.Logs(ctx)
+		if err != nil {
+			return fmt.Errorf("Timed out awaiting container, and failed to obtain container logs: %w", err)
+		}
+		if bytes, err := ioutil.ReadAll(reader); err != nil {
+			_, _ = fmt.Fprint(os.Stderr, "=====================================\n")
+			_, _ = fmt.Fprint(os.Stderr, "Container logs\n")
+			_, _ = fmt.Fprint(os.Stderr, "=====================================\n")
+			_, _ = fmt.Fprintf(os.Stderr, "%s", bytes)
+
+			return fmt.Errorf("Timed out awaiting container")
+		}
+
+		return fmt.Errorf("Timed out awaiting container, but failed to read container logs: %w", err)
+	}
+}
+
+func (s zeebeWaitStrategy) waitForTopology(finishedChan chan error, zbClient zbc.Client) {
```

Comment:
If this method stays the same, one thing we can do here is have the parameter as `finished chan<- error` which makes this a write-only channel and explicitly tells us what happens with it.

Reply:
That's probably a good idea, we can make it bidi if we ever need it :+1: